extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -35,21 +35,19 @@ import { ISystemDatabaseReferenceSettings, IDacpacReferenceSettings, IProjectRef
  * Controller for managing project lifecycle
  */
 export class ProjectsController {
-	private projectTreeViewProvider: SqlDatabaseProjectTreeViewProvider;
 	private netCoreTool: NetCoreTool;
 	private buildHelper: BuildHelper;
 
 	projects: Project[] = [];
 	projFileWatchers = new Map<string, vscode.FileSystemWatcher>();
 
-	constructor(projTreeViewProvider: SqlDatabaseProjectTreeViewProvider) {
-		this.projectTreeViewProvider = projTreeViewProvider;
+	constructor() {
 		this.netCoreTool = new NetCoreTool();
 		this.buildHelper = new BuildHelper();
 	}
 
-	public refreshProjectsTree() {
-		this.projectTreeViewProvider.load(this.projects);
+	public refreshProjectsTree(treeProvider: vscode.TreeDataProvider<any>) {
```

Comment:
I'd prefer to accept a workspaceTreeItem as the parameter, shortly I'll be working on partial tree refreshing, knowing the tree item would be helpful. also, you don't have to do context.treeDataProvider when calling this method :)